### PR TITLE
place layer changes from OMT

### PR DESCRIPTION
**Pull request by mmc1718**
- Copied changes from https://github.com/ACMEAtronOmatic/openmaptiles/blob/master/layers/place/city.sql

### CODE REVIEW

File: planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java [hunk 1]
```diff
@@ -108,10 +108,7 @@ public class Place implements
   private static final double MAX_CITY_POPULATION = 100_000_000d;
   private static final Set<String> MAJOR_CITY_PLACES = Set.of("city", "town", "village");
   private static final ZoomFunction<Number> LABEL_GRID_LIMITS = ZoomFunction.fromMaxZoomThresholds(Map.of(
-    8, 4,
-    9, 8,
-    10, 12,
-    12, 14
+    12, 15
```
Comment (nmr8acme) [inline planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java:111]: I don't understand what's going on here, can you explain it?

File: planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java [hunk 2]
```diff
@@ -340,7 +337,7 @@ public void process(Tables.OsmCityPoint element, FeatureCollector features) {
 
     int minzoom = rank != null && rank == 1 ? 2 :
       rank != null && rank <= 8 ? Math.max(3, rank - 1) :
-      placeType.ordinal() <= PlaceType.TOWN.ordinal() ? 7 :
+      placeType.ordinal() <= PlaceType.TOWN.ordinal() ? 4 :
```
Comment (nmr8acme) [inline planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java:340]: Can you add a link from this PR's description or from a commit message in this PR back to the PR or commit in the openmaptiles repo which made the same label change?

File: planetiler-basemap/src/test/java/com/onthegomap/planetiler/basemap/layers/PlaceTest.java [hunk 3]
```diff
@@ -473,6 +485,49 @@ void testCityWithoutNaturalEarthMatch() {
       ))));
   }
 
+
+    @Test
+  void testTown() {
```
Comment (nmr8acme) [inline planetiler-basemap/src/test/java/com/onthegomap/planetiler/basemap/layers/PlaceTest.java:490]: 👍

File: planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java (hunk 2 above)
Comment (mmc1718) [reply planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java:340]: https://github.com/ACMEAtronOmatic/openmaptiles/pull/1/commits/6bb701b7b26055238552c2553c453625366b47bd

Comment (nmr8acme) [reply planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java:340]: Also: https://github.com/ACMEAtronOmatic/openmaptiles/pull/1

File: planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java (hunk 1 above)
Comment (mmc1718) [reply planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/layers/Place.java:111]: Replicates https://github.com/ACMEAtronOmatic/openmaptiles/pull/1/commits/020cc15aefdeedb700013c9d9841efa4895c3fb5

Explanation:
This function maps zoom levels to 'label limits'. The label limits are used to determine how many labels are allowed to appear in a given square of the tile per zoom (the tile is split into a grid). This only applies to places that do not have a rank, i.e. are not found in Natural Earth data. See line https://github.com/nmr8acme/planetiler/pull/1/files#diff-8e85b00e58d492c0d11ae5481a17f671b0831a1c41c157c695b01a7749f06a5fR353 

By deleting the limits up to zoom 12, the limit of number of place labels per square on the grid is increased, so more places show. In the SQL version this was achieved by assigning a 'gridrank' number to each place.

It took me a while to figure out what was happening here, hopefully that makes sense

**Pull request merged**